### PR TITLE
Add Redux

### LINKED ISSUE #97: Add Redux to Manage State

### PR DESCRIPTION
**Pull request by nchouard**
Closes #97 

I borrowed a lot from the Redux docs and examples (especially the "Real World" example they have), so most of the design decisions come from working with that. Always interested in how the design could be improved!

### CODE REVIEW

File: penny_university_frontend/src/index.js [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+import App from './App'
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const store = createStore(
+  rootReducer,
+  applyMiddleware(thunk, api)
+)
+
+ReactDOM.render(
+  <Router>
```
[the remark this answers is not in the document]
Comment (nchouard) [reply penny_university_frontend/src/index.js]: I don't actually know. This is how it was set up in the tutorials so I was inclined to follow that structure.

File: penny_university_frontend/src/components/buttons/HeartButton.js [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-import PropTypes from 'prop-types';
+import React from 'react'
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
+import {faHeart} from '@fortawesome/free-solid-svg-icons'
+import {Button} from 'reactstrap'
+import PropTypes from 'prop-types'
 
 const HeartButton = ({className, count, size}) => {
   return (
     <Button outline size={size} color='danger' className={className}>
```
[the remark this answers is not in the document]
Comment (nchouard) [reply penny_university_frontend/src/components/buttons/HeartButton.js]: That's really just how bootstrap names its colors. I could create another class that just inherits the danger class with a different name if that is preferable.

File: penny_university_frontend/src/containers/ChatsPage.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  return {
+    filteredChats,
+    nextPageUrl
+  }
+}
+
+export default withRouter(connect(mapStateToProps, {
+  loadChatsList
+})(ChatsPage))
```
[the remark this answers is not in the document]
Comment (nchouard) [reply penny_university_frontend/src/containers/ChatsPage.js]: Yep!

File: penny_university_frontend/src/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nchouard) [reply penny_university_frontend/src/index.js]: I don't actually know. This is how it was set up in the tutorials so I was inclined to follow that structure.

File: penny_university_frontend/src/components/buttons/HeartButton.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JnBrymn) [reply penny_university_frontend/src/components/buttons/HeartButton.js]: I don't know CSS.

Inheriting from "danger" to make the color red seems weird. Having a standard red color defined for our website and then making the heart have that color seems sensible. But I don't know how to do that.

But also, I don't really care. FG=0.07

**Pull request merged**